plugins/incremental-ingestion-backend/src/routes.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorHandler } from '@backstage/backend-common';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { IncrementalIngestionDatabaseManager } from './database/IncrementalIngestionDatabaseManager';
+
+/** @public */
+export const createIncrementalProviderRouter = async (
+  manager: IncrementalIngestionDatabaseManager,
+  logger: Logger,
+) => {
+  const router = Router();
+  router.use(express.json());
+
+  // Get the overall health of all incremental providers
+  router.get('/health', async (_, res) => {
+    const records = await manager.healthcheck();
+    const providers = records.map(record => record.provider_name);
+    const duplicates = [
+      ...new Set(providers.filter((e, i, a) => a.indexOf(e) !== i)),
+    ];
+
+    if (duplicates.length > 0) {
+      res.json({ healthy: false, duplicateIngestions: duplicates });
+    } else {
+      res.json({ healthy: true });
+    }
+  });
+
+  // Clean up and pause all providers
+  router.delete('/cleanup', async (_, res) => {
```

Comment:
nah this should just be POST

Reply:
Yeah, I was trying to be fancy with it, but the `/cleanup` route is never going to have enough uses to warrant using `DELETE` for one of them.